### PR TITLE
GeoMesa 1.2.5 Update

### PR DESCRIPTION
**Pull request by jamesmcclain**
Update index/querying notes and feature list to reflect changes made in GeoMesa 1.2.5.

Still Needs:
   - [x] Approval of GeoMesa team

### CODE REVIEW

**Comment by lossyrob** [comment]
Awaiting comments from GeoMesa team.

File: GeoMesaQuery.md [hunk 1]
```diff
@@ -22,13 +23,18 @@ but can be configured to hold the [day, month, or year](https://github.com/locat
...
 For points, X and Y both have [31 bits](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-z3/src/main/scala/org/locationtech/geomesa/curve/Z2SFC.scala#L16-L17) of precision.
-Objects with extent are stored with [22 total bits](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/Z2Table.scala#L38-L43).
+
+### XZ3 and XZ2 ###
+
+XZ3 and XZ2 are the [default indices used to store objects with extent in GeoMesa 1.2.5](https://geomesa.atlassian.net/wiki/display/GEOMESA/GeoMesa+1.2.5+Release+Notes).
+An extension of Z-order, [XZ-order](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.73.4894) enlarges regions associated with nodes at the same level within the tree -- the tree implied by the pattern of divisions -- so that they overlap.
+This allows items to be associated with only one (Z-order) address,
+because the boundaries between neighboring nodes are now rectangles rather than lines,
+which resolves the problem of having to multiply-store of objects with extent.
```
Comment (elahrvivaz) [inline GeoMesaQuery.md]: if you're looking for number of bits - we use a length of 12 for both xz curves. That translates to 24 bits for xz2 and 36 bits for xz3.

Comment (jamesmcclain) [inline GeoMesaQuery.md]: Okay, I made that note in the feature list, but failed to make it here.  I will update.

**Comment by jamesmcclain** [comment]
@elahrvivaz Thank you such much for your feedback, it is very much appreciated!

**Comment by lossyrob** [comment]
+1

**Pull request merged**